### ISSUE TITLE
Provide an easy way to format lint messages with user-specified width

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#1884]
No matter where lint messages are printed (e.g. GitHub annotations, RStudio markers, Markdown in GitHub issues, etc.), currently, there is no straightforward way to print them with a certain width. This means that the user always needs to scroll horizontally to see many of the lint messages. 

``` r
options(width = 50)
library(lintr)

lint(
  text = "x <- c();",
  linters = unnecessary_concatenation_linter()
)
#> <text>:1:6: style: [unnecessary_concatenation_linter] Unneeded concatenation without arguments. Replace the "c" call by NULL or, whenever possible, vector() seeded with the correct type and/or length.
#> x <- c();
#>      ^~~
```

<sup>Created on 2023-01-03 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

A typical project that starts using linters might throw hundreds of messages, and that's a lot of scrolling. This might be mildly annoying for an able-bodied individual like me, but must be extremely challenging and disappointing for people with mobility impairment. Almost every single lint asks them to scroll to see the full lint message. 

We should anticipate this and provide a way out.

**Comment by IndrajeetPatil** [comment]
With #2369:

``` r
options(lintr.format_width = 80)
library(lintr)

lint(
  text = "x <- c();",
  linters = unnecessary_concatenation_linter()
)
#> <text>:1:6: style: [unnecessary_concatenation_linter] Unneeded concatenation
#>     without arguments. Replace the "c" call by NULL or, whenever possible,
#>     vector() seeded with the correct type and/or length.
#> x <- c();
#>      ^~~
```

<sup>Created on 2023-11-30 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
